angular/core/components/thread_preview/thread_preview.coffee

```diff
@@ -3,10 +3,23 @@ angular.module('loomioApp').directive 'threadPreview', ->
   restrict: 'E'
   templateUrl: 'generated/components/thread_preview/thread_preview.html'
   replace: true
-  controller: ($scope, Records, Session, LmoUrlService, FlashService, ModalService, MuteExplanationModal) ->
+  controller: ($scope, Records, Session, LmoUrlService, FlashService, ModalService, MuteExplanationModal, DismissExplanationModal) ->
     $scope.lastVoteByCurrentUser = (thread) ->
       thread.activeProposal().lastVoteByUser(Session.user())
 
+    $scope.dismiss = ->
+      if !Session.user().hasExperienced("dismissThread")
+        Records.users.saveExperience("dismissThread")
+        ModalService.open DismissExplanationModal, thread: -> $scope.thread
+      else
+        $scope.thread.dismiss()
+        FlashService.success "dashboard_page.thread_dismissed"
+
+    $scope.undismiss = ->
+      $scope.thread.dismiss()
```

Comment:
?? Is dismiss a toggleable function? Should it be named differently in that case?